kie-wb-common-react/boxed-expression-editor/src/components/ExpressionContainer/ExpressionContainer.tsx

```diff
@@ -74,44 +84,67 @@ export const ExpressionContainer: (props: ExpressionContainerProps) => JSX.Eleme
         ]}
       />
     );
-  };
+  }, [i18n.clear, actionDropdownIsOpen, logicTypeIsPresent, executeClearAction]);
+
+  const getLogicTypesWithoutUndefined = useCallback(() => {
+    return Object.values(LogicType).filter((logicType) => logicType !== LogicType.Undefined);
+  }, []);
+
+  const renderLogicTypeItems = useCallback(() => {
+    return _.map(getLogicTypesWithoutUndefined(), (key) => <SimpleListItem key={key}>{key}</SimpleListItem>);
+  }, [getLogicTypesWithoutUndefined]);
 
-  const buildLogicSelectorMenu = () => {
+  const buildLogicSelectorMenu = useCallback(() => {
     return (
       <PopoverMenu
         title={i18n.selectLogicType}
-        arrowPlacement={() => document.getElementById("expression-container-box")!}
+        arrowPlacement={() => document.querySelector(".expression-container-box")! as HTMLElement}
         body={<SimpleList onSelect={onLogicTypeSelect}>{renderLogicTypeItems()}</SimpleList>}
       />
     );
-  };
-
-  const renderLogicTypeItems = () => {
-    return _.map(
-      [
-        i18n.literalExpression,
-        i18n.context,
-        i18n.decisionTable,
-        i18n.relation,
-        i18n.function,
-        i18n.invocation,
-        i18n.list,
-      ],
-      (key) => <SimpleListItem key={key}>{key}</SimpleListItem>
-    );
-  };
+  }, [i18n.selectLogicType, onLogicTypeSelect, renderLogicTypeItems]);
+
+  const updateNameAndDataType = useCallback(
+    (updatedName, updatedDataType) => {
+      setSelectedExpression({
+        ...selectedExpression,
+        name: updatedName,
+        dataType: updatedDataType,
+      });
+    },
+    [selectedExpression]
+  );
+
+  const renderSelectedExpression = useCallback(
+    (selectedExpression: ExpressionProps) => {
+      selectedExpression.updateNameAndDataTypeCallback = updateNameAndDataType;
```

Comment:
This smells bad to me. You're updating a state using a callback instead of changing the state, this is not a good practice and may bite you in the future 😝